abi-versions/README.md

```diff
@@ -0,0 +1,40 @@
+# Proxy-Wasm ABI specification
+
+## ABI versioning
+
+The ABI is in the form of: `<major>.<minor>.<patch>`
+
+Breaking changes require increasing the major version.
+
+Adding new functions requires increasing the minor version.
+
+Adding new enum values requires increasing the patch version.
+
```

Comment:
One high level question I have is how standardized everything is going to be with WASM. Will we be able to add new callbacks or methods below as the need arises in Envoy extension development, or will this require some external process? Ideally we can review a MVP ABI here and grow it as needed, rather than trying to account for all possible use cases.

Reply:
Ideally, those new callbacks and/or host functions would be generic enough that they would be added to this ABI.

There is a note in the paragraph below about implementation-specific extensions (e.g. `envoy_`), but that's really an escape hatch and hopefully we shouldn't need anything like that in practice.

I fully agree regarding MVP approach. Everything that you see in this ABI was added during Istio Telemetry v2 development, we didn't enumerate all possible cases or add anything that's not already implemented.

Reply:
I think adding a section calling this out would be helpful. E.g. call it "ABI extension" and describe how FFI is the first expected phase or escape hatch, then the process for having a more generic extension of host functions/callbacks added via WG. What's the process, etc. Which repositories, which PRs, etc.